roles/confluent.common/tasks/remove_packages.yml

```
@@ -26,11 +26,16 @@
     msg: "{{ confluent_packages_removed }}"
   when: confluent_packages_removed|length > 0
 
+### Kafka Connect Replicator Binary does not include a service definition by default, thus we need to validate if one exists in order to handle service lifecycle correctly.
+### - ansible_facts.services[service_name + '.service'] is defined
```

```suggestion
- name: Populate service facts
  service_facts:
```
WDYT about keeping this task just in case a user creates a custom playbook calling our component roles? I know of at least one user creating his own playbooks. If the previous `service_facts` invocation was within our roles (and not the playbooks) then I'd feel more comfortable removing it here. (sorry I changed my mind on this one lol)

---

Why remove the comment?  I think it makes sense to leave it as it's not obvious why we need the check even if we populate the services facts again.

I also think your point about removing the service facts makes sense as this isn't designed to be a self contained role in that sense, and this is extraneous code otherwise.

---

Oh sorry, keep the comment if you prefer. That's a good idea.

Still, I think roles should work though regardless of which playbooks call them. A good example why is our `schema_registry.yml` does not call the service_facts module: 
https://github.com/confluentinc/cp-ansible/blob/6.1.x/schema_registry.yml

Like out others do:
https://github.com/confluentinc/cp-ansible/blob/6.1.x/control_center.yml#L11

So in the case of SR, we need this task

---

I totally agree with @domenicbove on this as a general principle. All roles should ideally be self-contained, and useable on their own - not relying on encapsulating playbooks. Playbooks could add functionality/features, like the serial processing for kafka_broker and zookeeper host, but no role should **require** to be run by a particular playbook.....